Plugins/BuildServerIntegration/AzureDevOpsIntegration/AzureDevOpsAdapter.cs

```diff
@@ -36,6 +35,12 @@ internal class AzureDevOpsAdapter : IBuildServerAdapter
         private IntegrationSettings _settings;
         private ApiClient _apiClient;
         private static readonly IBuildDurationFormatter _buildDurationFormatter = new BuildDurationFormatter();
+        private Task<string> _buildDefinitionsTask;
+        private string _projectUrl;
+        private static CacheAzureDevOps CacheAzureDevOps = null;
```

Comment:
This is static because a new instance of this class is created after each revision grid refresh.
The goal is here to keep some data that won't change in cache to keep it for the next instance created and so avoid one or two api calls.

So, there is no synchronization problems (because the instance is only in concurrence with futur herself ![](https://media.giphy.com/media/xsF1FSDbjguis/source.gif) and will always access it first without problems 😉  ) and I decided to keep the code simple.

Reply:
Hahah nice gif

Reply:
Ok then, it should be documented in a form of a comment.
As it isn't obvious on its own.

Reply:
Done